util.go

```diff
@@ -34,10 +34,9 @@ func (d Date) AddDate(year, month, day int) Date {
 func (d Date) AddMonths(n int) Date {
 	year, month, day := d.YMD()
 	iMonth := int(month) + n
-	if iMonth <= 0 {
-		iMonth -= _monthsOfYear
+	if iMonth > _monthsOfYear {
+		year += iMonth / _monthsOfYear
```

Comment:
`if iMonth <= 0 {` in case the n is negative (we want to SubMonths). 

I think the issue here is when iMonth = 12 * k.